action/candidate_register.go

```diff
@@ -289,6 +293,9 @@ func (cr *CandidateRegister) SanityCheck() error {
 	if cr.Amount().Sign() <= 0 {
 		return errors.Wrap(ErrInvalidAmount, "negative value")
 	}
+	if err := validator.ValidateCandidateNameForStake2(cr.Name()); err != nil {
+		return errors.Wrap(ErrInvalidCandidateName, err.Error())
+	}
```

Comment:
just to be sure, this change will be called by `actpool` when accepting taking in a tx right?

if that's the case, need to gate this by next HF for safety, since node will take in or discard this action with or w/o this check

Reply:
Yes I  debugged
```
github.com/iotexproject/iotex-core/action.(*CandidateRegister).SanityCheck
	/iotex/iotex-core/action/candidate_register.go:290
github.com/iotexproject/iotex-core/action/protocol.(*GenericValidator).Validate
	/iotex/iotex-core/action/protocol/generic_validator.go:74
github.com/iotexproject/iotex-core/actpool.(*actPool).validate
	/iotex/iotex-core/actpool/actpool.go:356
github.com/iotexproject/iotex-core/actpool.(*actPool).Add
	/iotex/iotex-core/actpool/actpool.go:229
github.com/iotexproject/iotex-core/api.(*coreService).SendAction
	/iotex/iotex-core/api/coreservice.go:405
```

Reply:
also, change to use `isValidCandidateName()`, don't use ioctl package